js/Config.js

```diff
@@ -0,0 +1,74 @@
+function FC() {
+	this._subConfigCategories = [];
+	
+	this.setConfigProvider = function(configProvider) {
+		this._subConfigCategories.forEach(function (subConfig) {
+			subConfig.setConfigProvider(configProvider);
+		});	
+	}
+	this.add = function(config) {
+		this._subConfigCategories.push(config);
+	}
+	this.purge = function()  {
+	    this._subConfigCategories.forEach(function (subConfig) {
+            subConfig.refresh();
+        }); 
+	}
+}
+
+function Config(properties) {
+	this.properties = properties;
+	this.propertyRefreshed = {};
+	this._configProvider;
+	this._listeners = [];
+	
+	Object.keys(properties).forEach(function (prop) {	
+		Object.defineProperty(this, prop, {
+			get: function () {
+				if (this.propertyRefreshed[prop] || prop === 'configName')
+					return properties[prop];
+				else {
+					try {
+						this.requestUpdate(prop);
+						return properties[prop];
+					} catch (e) {
+						return null;
```

Comment:
We shouldn't just swallow exceptions. This will most likely break in most cases.